src/function.cpp

```diff
@@ -46,6 +49,31 @@ DeclaredFunction DeclaredFunction::declare(std::vector<mlir::Type> &&domain,
     throw UnsupportedException("Unsupported type: " + to_string(t));
   };
 
+  if (rangeDimRefIdx) {
+    const auto dimRefIdx = *rangeDimRefIdx;
+    smart_assert(dimRefIdx < domain.size(),
+                 "Tried to refer to the argument of an invalid index");
+
+    const auto shapedDimRef = domain[dimRefIdx].dyn_cast<mlir::ShapedType>();
+    const auto shapedRange = range.dyn_cast<mlir::ShapedType>();
```

Comment:
Is const necessary..?

Reply:
I tend to declare the const variables unless the variable must be mutated. Would it be unnecessary though?

Reply:
Hmm, makes sense!